fast_matched_filter/src/matched_filter.c

```diff
@@ -474,6 +564,45 @@ void network_corr_precise_no_sum(
     }
 }
 
+float network_corr_variable_precise(
+    float *templates, float *sum_square_template, int *moveouts,
+    float *data, float *weights, float *cc_norm, int *n_samples_template, size_t max_samples_template,
+    size_t n_samples_data, size_t n_stations, size_t n_components, int normalize)
+{
+
+    size_t d, t;
+    size_t station_offset, component_offset;
+    float cc, cc_sum = 0; // output
+
+    for (size_t s = 0; s < n_stations; s++)
+    {
+
+        station_offset = s * n_components;
+
+        cc = 0;
+        for (size_t c = 0; c < n_components; c++)
+        {
+            component_offset = station_offset + c;
+            if (weights[component_offset] == 0)
+                continue;
+
+            // if ((i + moveouts[component_offset]) > n_samples_data - n_samples_template) continue;
+
+            t = component_offset * max_samples_template;
+            d = component_offset * n_samples_data + moveouts[component_offset];
+
+            cc = corrc_precise(templates + t,
+                               sum_square_template[component_offset],
+                               data + d,
+                               n_samples_template[component_offset],
+                               normalize);
+            cc_sum += atanhf(cc - FLT_EPSILON) * cc_norm[component_offset];
```

Comment:
I admit I don't remember why I did this a long time ago when I originally wrote this up. Thinking about it now, it does make sense to remove a `FLT_EPSILON` and then add it back in to avoid any `INF` from a potential `atanh(1)` (perfect correlation with the template itself).

That said, if one really wanted to make it resistant to `atanh(-1)`, one could think about carrying around what was added or removed in some error variable, using the sign of cc to decide whether to remove or add a `FLT_EPSILON`.

For the vast majority of cases, I don't think we're going to run into -1.000000 correlations, so I won't tackle that now.

Though I do think the current implementation isn't correct because it removes a `FLT_EPSILON` from every `cc`, but only adds back on `FLT_EPSILON` for the sum. I think it should add back `n_stations * n_components * FLT_EPSILON`, no?

I'll make this change shortly and double check that it works.

Reply:
So I made this happen. You'll see I included `10 * FLT_EPSILON` because I was sometimes getting `NaN`'s with just one. The results are still not as precise as the precise implementation, but I'm not quite sure what to do at this point.